radar_parlamentar/importadores/conv.py

```diff
@@ -118,124 +153,18 @@ def _gera_votos(self, votacao, nome_partido, opcoes):
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_PRIMEIRO_SEMESTRE, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.ABSTENCAO, models.NAO]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.SIM, models.NAO, models.AUSENTE]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
-
-    def _gera_votacao5(self):
-
-        numero_proposicao = '5'
-        descricao_proposicao = 'Guilhotinar o Conde Pierre'
-        prop = self._gera_proposicao(numero_proposicao, descricao_proposicao)
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_SEGUNDO_SEMESTRE, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.ABSTENCAO]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.NAO, models.NAO, models.NAO]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
-
-    def _gera_votacao6(self):
-
-        numero_proposicao = '6'
-        descricao_proposicao = 'Criação de novas escolas'
-        prop = self._gera_proposicao(numero_proposicao, descricao_proposicao)
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_SEGUNDO_SEMESTRE, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.AUSENTE, models.SIM, models.SIM]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
-
-    def _gera_votacao7(self):
-
-        numero_proposicao = '7'
-        descricao_proposicao = 'Aumento do efetivo militar'
-        prop = self._gera_proposicao(numero_proposicao, descricao_proposicao)
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_SEGUNDO_SEMESTRE, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.ABSTENCAO]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.SIM, models.AUSENTE, models.SIM]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
+    def _gera_votacao_geral(self):
+        for i in range(len(self._votacao_params)):
```

Comment:
Pessoal,

primeiro que não faz sentido usar um dicionário (`_votacao_params`) e tratar as "keys" do dicionário como "índices".

Segundo que ao invés de usar `for i in range(len(self._votacao_params))` e depois ficar acessando os "índices", podemos ser mais diretos com python e "loopar" diretamente nos itens.

Exexemplo didático caso `_votacao_params` seja um dicionário:
```python
_votacao_params = {
    1: ['Reforma agrária',
         [models.SIM, models.ABSTENCAO, models.NAO],
         [models.SIM, models.SIM, models.SIM],
         [models.NAO, models.NAO, models.NAO]],
    2: ['Aumento da pensão dos nobres',
         [models.NAO, models.NAO, models.NAO],
         [models.NAO, models.NAO, models.NAO],
         [models.SIM, models.SIM, models.SIM]]
}

for key, value in _votacao_params.items():
    list_params = value
    numero_prop = int(key)
    descricao_prop = list_params[0]
    prop = self._gera_proposicao(numero_prop, descricao_prop)
    votacao = self._gera_votacao(
        numero_prop, descricao_prop,
        DATA_NO_PRIMEIRO_SEMESTRE, prop)
    self._gera_votos(votacao, GIRONDINOS, list_params[1])
    self._gera_votos(votacao, JACOBINOS, list_params[2])
    self._gera_votos(votacao, MONARQUISTAS, list_params[3])
```

Outra opção é usar uma lista no lugar do dicionário:
```python
_votacao_params = [
    ['Reforma agrária',
     [models.SIM, models.ABSTENCAO, models.NAO],
     [models.SIM, models.SIM, models.SIM],
     [models.NAO, models.NAO, models.NAO]
    ],
    ['Aumento da pensão dos nobres',
     [models.NAO, models.NAO, models.NAO],
     [models.NAO, models.NAO, models.NAO],
     [models.SIM, models.SIM, models.SIM]
    ]
]

for index, item in enumerate(_votacao_params):
    list_params = item
    numero_prop = int(index+1)
    descricao_prop = list_params[0]
    prop = self._gera_proposicao(numero_prop, descricao_prop)
    votacao = self._gera_votacao(
        numero_prop, descricao_prop,
        DATA_NO_PRIMEIRO_SEMESTRE, prop)
    self._gera_votos(votacao, GIRONDINOS, list_params[1])
    self._gera_votos(votacao, JACOBINOS, list_params[2])
    self._gera_votos(votacao, MONARQUISTAS, list_params[3])
```

Mas, objetivando ficar com apenas 1 método "gera_votacoes", eu faria algo mais ou menos na seguinte linha:
```python
_votacao_params = [
    {"descricao": 'Reforma agrária',
     "partidos": {
         "GIRONDINOS": [models.SIM, models.ABSTENCAO, models.NAO],
         "JACOBINOS": [models.SIM, models.SIM, models.SIM],
         "MONARQUISTAS": [models.NAO, models.NAO, models.NAO])
]

for index, votacao in enumerate(_votacao_params):
    numero = int(index+1)
    descricao = votacao.get("descricao")
    proposicao = self._gera_proposicao(numero, descricao)
    partidos = votacao.get("partidos")
    votacao = self._gera_votacao(
        numero, descricao, DATA_NO_PRIMEIRO_SEMESTRE, proposicao)
    for partido, votos in partidos.items():
        self._gera_votos(votacao, partido, votos)
```

Dessa forma vocês conseguem inclusive incluir os dados da "votação8" e remover também o método "gera_proposicao10".